### PR TITLE
Ensure trailing slashes in paths are preserved

### LINKED ISSUE #195: Trailing '/' is removed from paths
I believe the current (1.1.0 in `master`) URL encoding mechanism drops trailing `/` from the path:
https://github.com/brendanhay/amazonka/blob/develop/core/src/Network/AWS/Data/Path.hs#L50-L51:
```haskell
rawPath :: ToPath a => a -> Path 'NoEncoding
rawPath = Raw . filter (not . BS8.null) . BS8.split sep . toPath
``` 
This function converts paths like this:
```
abc/def/
```
Into this:
```
abc/def
```

As a result S3 service does not work correctly with otherwise valid object keys (e.g. keys with trailing `/`'s).

### PR DESCRIPTION
**Pull request by brendanhay**
Ensures erroneous paths are corrected, but trailing slashes are retained: `/foo///bar/` becomes `/foo/bar/`.

Fixes #195.

### CODE REVIEW

File: core/src/Network/AWS/Types.hs [hunk 1]
```diff
@@ -604,17 +603,16 @@ newtype Seconds = Seconds Int
         , ToText
         )
 
-_Seconds :: Iso' Seconds Int
-_Seconds = iso seconds Seconds
-
 instance ToLog Seconds where
-    build (Seconds n) = build n <> "s"
+    build s = build (seconds s) <> "s"
```
Comment (brendanhay) [inline core/src/Network/AWS/Types.hs]: What the. I must have been sleep coding!

**Comment by brendanhay** [comment]
In hindsight, I'm not 100% certain about the decision to correct the adjacent slashes within the path. I'll leave it for now, since it's only the trailing slash behaviour that has been changed, but may revisit.

**Comment by EduardSergeev** [comment]
@brendanhay Thank you for fixing it! My understanding that at least for `S3` adjacent slashes is not an error. Ideally (for us) the S3 path (object key) should not be changed at all. I suspect that for other services this slash handling may be required, but for `S3` it doesn't seem to be.

**Comment by brendanhay** [comment]
Yeah, agreed. I'll take care of it separately. Thanks for raising the issue.

**Pull request merged**